harambe/parser/parser.py

```diff
@@ -50,6 +56,12 @@ def validate(self, data: dict[str, Any], base_url: URL) -> dict[str, Any]:
         self.field_types = self._get_field_types(base_url)
         self.model = self._schema_to_pydantic_model(self.schema)
         cleaned_data = trim_keys_and_strip_values(data)
+        all_required_fields = self._get_all_required_fields(self.schema)
+        missing_fields = self._find_missing_required_fields(
+            cleaned_data, all_required_fields
+        )
+        if missing_fields:
+            raise RequiredFieldsError(all_required_fields)
         if self._all_fields_empty(cleaned_data):
             raise SchemaValidationError(
```

Comment:
We should raise a different kind of Schema Validation Error here. One that doesn't need to display all of the empty data. This will make it easier to parse through logs (its a bit cloudy right now). Thoughts @awtkns

Reply:
lets think about it in a separate ticket